test/blackbox-tests/test-cases/promote/promotion-show.t

```diff
@@ -0,0 +1,257 @@
+Test the 'dune promotion show' command
+
+=== WHAT THIS TEST VERIFIES ===
+This test ensures that 'dune promotion show' correctly displays the contents
+of corrected files that are ready for promotion, without modifying the source
+files. The command allows users to preview what changes would be applied before
+running 'dune promote'.
+
+Key properties tested:
+1. Shows corrected content for files with failed diff rules
+2. Is read-only (never modifies source files)
+3. Handles single file, multiple files, and all files (no args)
+4. Proper warning handling for non-promotable files
+5. Files with 'diff' (not 'diff?') are promotable
+
+=== WHY THIS MATTERS ===
+The 'show' command is critical for user workflow: it lets developers review
+auto-generated corrections before accepting them. This is especially important
+for generated code, test expectations, or formatted output where users need to
+verify the changes make sense before committing them.
+
+=== HOW TO VERIFY CORRECTNESS ===
+1. Before 'runtest': show should warn (no corrections available)
+2. After 'runtest': show should display corrected content from _build
+3. After 'show': source files should remain unchanged (read-only guarantee)
+4. Output includes trailing newline after each file's content
+5. Only 'diff' rules create promotions; 'diff?' rules may not always generate them
+
+Setup a basic dune project with promotion-capable rules
+
+We create different promotion scenarios to test various behaviors:
+- a.expected: uses 'diff' (creates promotion when files differ)
+- b.expected: uses 'diff?' in progn (creates promotion when files differ)
+- c.expected: uses 'diff?' in separate action (may not create promotion)
+
+  $ cat > dune-project << EOF
+  > (lang dune 2.0)
+  > EOF
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias runtest)
+  >  (action
+  >   (diff a.expected a.actual)))
+  > 
+  > (rule
+  >  (with-stdout-to a.actual
+  >   (echo "A actual\n")))
+  > 
+  > (rule
+  >  (alias runtest)
+  >  (action
+  >   (progn
+  >    (with-stdout-to b.actual
+  >     (echo "B actual\n"))
+  >    (diff? b.expected b.actual))))
+  > 
+  > (rule
+  >  (with-stdout-to c.actual
+  >   (echo "C actual\n")))
+  > 
+  > (rule
+  >  (alias runtest)
+  >  (action
+  >   (diff? c.expected c.actual)))
+  > EOF
+
+  $ echo 'A expected' > a.expected
+  $ echo 'B expected' > b.expected
+  $ echo 'C expected' > c.expected
+
+=== TEST: Before tests run, no promotions available ===
+When no tests have run yet, there are no corrected files in _build,
+so 'show' should return a warning (not an error, as this is informational).
+
+  $ dune promotion show a.expected 2>&1
```

Comment:
It's a nitpick, but these redirections `2>&1` don't need to be there, everything is printed in the cram output anyway